packages/twenty-server/src/engine/core-modules/auth/services/google-apis.service.ts

```diff
@@ -114,7 +121,27 @@ export class GoogleAPIsService {
           manager,
         );
 
-        await messageChannelRepository.save(
+        const connectedAccountMetadata =
```

Comment:
First read:
- Shouldn't we also check the workspaceId?

Second read: (optional)
i think it's fine as we are in the write path but we could also leverage the cached metadata (through WorkspaceCacheStorageService) to avoid performing additional queries on the database.

Third read: (optional)
I would actually remove objectMetadata from the Event payload completly:
- as we already have objectMetadataNameSingular as part of the BatchEvent. 
- Also looks a bit heavy to force the caller to provide the whole objectMetadata everytime. I think there is only a few places we will need to have more than the nameSingular (maybe timelineActivity need the id for instance), we could leave it to timelineActivity to fetch the objectMetadata based on the nameSingular

Reply:
Agree with the comments. I'm not sure how far I can go as part of this PR.
1 should be done for sure (I copied the query from somewhere else so I will check it's probably missing in other places). As this pattern is already used elsewhere, 2 and 3 might require large refactoring that shouldn't be part of this PR - in that case should I create a followup issue that details (3) and merge with (1) for now?

Reply:
yes LGTM